src/client.rs

```diff
@@ -626,3 +633,115 @@ where
         }
     }
 }
+
+impl Client<TonicClient> {
+    /// Encrypt binary data using Dapr. returns Vec<StreamPayload> to be used in decrypt method
+    ///
+    /// # Arguments
+    ///
+    /// * `payload` - A Reference to the data to encrypt, should impl Into<Vec<u8>>
+    /// * `request_option` - Encryption request options.
+    ///
+    /// # Example
+    /// ```
+    /// let encrypted = client
+    ///         .encrypt(
+    ///             &"Test".to_string(),
+    ///             EncryptRequestOptions {
+    ///                 component_name: "localstorage".to_string(),
+    ///                 key_name: "rsa-private-key.pem".to_string(),
+    ///                 key_wrap_algorithm: "RSA".to_string(),
+    ///                 data_encryption_cipher: "aes-gcm".to_string(),
+    ///                 omit_decryption_key_name: false,
+    ///                 decryption_key_name: "rsa-private-key.pem".to_string(),
+    ///             },
+    ///         )
+    ///         .await
+    ///         .unwrap();
+    /// ```
+    pub async fn encrypt<T>(
+        &mut self,
+        payload: &T,
+        request_options: EncryptRequestOptions,
+    ) -> Result<Vec<StreamPayload>, Status>
+    where
+        T: Into<Vec<u8>> + Clone,
```

Comment:
How does this implementation fare versus using a BufReader?
The go implementation has a great example of using streams rather than a reference to an object

Reply:
I can do that but I would need to know the size of the chunks. Do you know what the standard size is?

Reply:
@mikeee i was able to do it via ReadStream in tokio-utils. It works with string, strs, etc... but for files it has to be tokio::fs::File. Do you see that as an issue?

Reply:
I also did some research into BuffReader but it don't optimize the chunk sizes or even chunk at all for that matter. You have to do that manually.